packages/field-label/src/FieldLabel.ts

```diff
@@ -0,0 +1,126 @@
+
+import styles from './field-label.css.js';
+
+type AcceptsFocusVisisble = HTMLElement & { forceFocusVisible?(): void };
+
+/**
+ * @element sp-field-label
+ */
+export class FieldLabel extends SpectrumElement {
+    public static get styles(): CSSResultArray {
+        return [styles, asterickIconStyles];
+    }
+
+    static instanceCount = 0;
+
+    @property({ type: Boolean, reflect: true })
+    public disabled = false;
+
+    @property({ type: String })
+    public id = '';
+
+    @property({ type: String })
+    public for = '';
+
+    @property({ type: Boolean, reflect: true })
+    public required = false;
+
+    @property({ type: String, reflect: true, attribute: 'side-aligned' })
+    public sideAligned?: 'start' | 'end';
+
+    private target?: HTMLElement;
+
+    private handleClick(): void {
+        if (!this.target || this.disabled) return;
+        this.target.focus();
+        const parent = this.getRootNode() as ShadowRoot;
+        const target = this.target as AcceptsFocusVisisble;
+        const targetParent = target.getRootNode() as ShadowRoot;
+        const targetHost = targetParent.host as AcceptsFocusVisisble;
+        if (targetParent.isSameNode(parent) && target.forceFocusVisible) {
+            target.forceFocusVisible();
+        } else if (targetHost && targetHost.forceFocusVisible) {
+            targetHost.forceFocusVisible();
+        }
+    }
+
+    private async manageFor(): Promise<void> {
+        if (!this.for) {
+            return;
+        }
+        const parent = this.getRootNode() as HTMLElement;
+        const target = parent.querySelector(`#${this.for}`) as Focusable;
+        if (typeof target.updateComplete !== 'undefined') {
+            await target.updateComplete;
+        }
+        this.target = target.focusElement || target;
+        if (this.target) {
+            const targetParent = this.target.getRootNode() as HTMLElement;
+            if (targetParent.isSameNode(parent)) {
+                this.target.setAttribute('aria-labelledby', this.id);
+            } else {
+                this.target.setAttribute(
+                    'aria-label',
+                    (this.textContent || /* c8 ignore next */ '').trim()
+                );
+            }
+        }
+    }
+
+    protected render(): TemplateResult {
+        return html`
+            <label>
+                <slot></slot>
+                ${this.required
+                    ? html`
+                          <sp-icon class="requiredIcon asterick">
+                              ${AsteriskIcon({ hidden: true })}
+                          </sp-icon>
+                      `
+                    : html``}
+            </label>
+        `;
+    }
+
+    protected firstUpdated(changes: PropertyValues): void {
+        super.firstUpdated(changes);
+        if (!this.hasAttribute('id')) {
+            this.setAttribute(
+                'id',
+                `sp-field-label-${FieldLabel.instanceCount++}`
```

Comment:
this is the sort of code that will cause subtle conflicts if multiple `sp-field-label` classes can be instantiated, or if this is subclassed. Can you use the tag name directly here? That'd at least give some safety here.

Reply:
Great idea. Updated!